std/numeric.d

```diff
@@ -2572,27 +2572,55 @@ GapWeightedSimilarityIncremental!(R, F) gapWeightedSimilarityIncremental(R, F)
 
 /**
 Computes the greatest common divisor of $(D a) and $(D b) by using
-Euclid's algorithm.
+Euclid's or Stein's algorithm depending on compiler.
  */
 T gcd(T)(T a, T b)
 {
-    static if (is(T == const) || is(T == immutable))
-    {
-        return gcd!(Unqual!T)(a, b);
-    }
-    else
-    {
-        static if (T.min < 0)
+    version(DigitalMars) {
+        static if (is(T == const) || is(T == immutable))
         {
-            assert(a >= 0 && b >=0);
+            return gcd!(Unqual!T)(a, b);
```

Comment:
This part is common to both implementations, so pull it before the version test.